sidebar.js

```diff
@@ -2,65 +2,25 @@ class SidebarNav extends HTMLElement {
   constructor () {
     super()
     this.attachShadow({ mode: 'open' })
+    this.initSidebar()
+  }
 
-    // Create sidebar container
-    const sidebar = document.createElement('div')
-    sidebar.classList.add('sidebar')
-
-    // Branding
-    const homePage = document.createElement('a')
-    homePage.href = './'
-    homePage.classList.add('home-page-link')
-    const headerBranding = document.createElement('h1')
-    headerBranding.classList.add('header-branding')
-    headerBranding.textContent = 'Social Reader'
-    homePage.appendChild(headerBranding)
-    sidebar.appendChild(homePage)
-
-    // Controls
-    const controls = document.createElement('div')
-    controls.classList.add('controls')
-    const followingLink = document.createElement('a')
-    followingLink.href = './followed-accounts.html'
-    followingLink.textContent = 'Following · '
-
-    const followedCount = document.createElement('followed-count')
-    followingLink.appendChild(followedCount)
-
-    controls.appendChild(followingLink)
-    sidebar.appendChild(controls)
-
-    // Navigation
-    const nav = document.createElement('nav')
-    const themeSelector = document.createElement('theme-selector')
-    nav.appendChild(themeSelector)
-    const links = [
-      { href: './about.html', text: 'About' },
-      {
-        href: 'https://hypha.coop/dripline/announcing-dp-social-inbox/',
-        text: 'Social Inbox'
-      },
-      { href: 'https://distributed.press', text: 'Distributed Press' }
-    ]
-    links.forEach((linkInfo) => {
-      const a = document.createElement('a')
-      a.href = linkInfo.href
-      a.textContent = linkInfo.text
-      nav.appendChild(a)
-    })
-    sidebar.appendChild(nav)
+  async initSidebar () {
+    const response = await fetch('./sidebar.html')
```

Comment:
Instead of doing this in the component do it at the top level of the module so it's only done once in case we migrate to client side routing and the component can get initialized more than once.